### PR TITLE
Add "agent.useHostNetwork" and "agent.dnsPolicy" value to chart.

### PR DESCRIPTION
**Pull request by ts4z**
If agent.useHostNetwork is true, tell jaeger-agents to set hostNetwork:
true in their specs, allowing other containers on the host access to the
shared agent without having to know its name or location, just the
host's IP address.

agent.dnsPolicy will be propigated to the agents as the dnsPolicy for
the pod.  ClusterFirstWithHostNet is useful when useHostNetwork is true.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: incubator/jaeger/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 1.2.0
 description: A Jaeger Helm chart for Kubernetes
 name: jaeger
-version: 0.3.13
+version: 0.3.14
```
Comment (timstoop) [inline incubator/jaeger/Chart.yaml]: This adds new features, so please bump the minor.

**Review by timstoop — CHANGES_REQUESTED** [review]

File: incubator/jaeger/Chart.yaml (hunk 1 above)
Comment (ts4z) [reply incubator/jaeger/Chart.yaml]: ok, done. thanks!

**Comment by pavelnikolov** [comment]
lgtm

**Comment by timstoop** [comment]
Needs a merge from master, after which the tests will probably succeed.

**Comment by ts4z** [comment]
Thank you!

**Pull request merged**